RFS/src/test/java/com/rfs/cms/OpenSearchWorkCoodinatorTest.java

```diff
@@ -0,0 +1,175 @@
+    @MethodSource("provideGetResultTestArgs")
+    public void testWhenGetResult(
+            Supplier<AbstractedHttpClient.AbstractHttpResponse> responseSupplier,
+            DocumentModificationResult expectedTesult) throws Exception {
+
+        // Set up our test
+        MockHttpClient client = new MockHttpClient(responseSupplier);
+
+        // Run the test
+        DocumentModificationResult result;
+        try (var workCoordinator = new OpenSearchWorkCoordinator(client, 2, "testWorker")) {
+            result = workCoordinator.getResult(responseSupplier.get());
+        }
+
+        // Verify the result
+        Assertions.assertEquals(expectedTesult, result);
+
+    }
+
+    @Test
+    public void testWhenGetResultAndConflictThenIgnored() throws Exception {
+        // Set up our test
+        class ConflictResponse extends TestResponse {
+            @Override
+            public String getStatusText() {
+                return "conflict";
+            }
+
+            @Override
+            public int getStatusCode() {
+                return 409;
+            }
+        }
+
+        Supplier<AbstractedHttpClient.AbstractHttpResponse> responseSupplier = () -> new ConflictResponse();
+        MockHttpClient client = new MockHttpClient(responseSupplier);
+
+        // Run the test
+        DocumentModificationResult result;
+        try (var workCoordinator = new OpenSearchWorkCoordinator(client, 2, "testWorker")) {
+            result = workCoordinator.getResult(responseSupplier.get());
+        }
+
+        // Verify the result
+        Assertions.assertEquals(DocumentModificationResult.IGNORED, result);
+
+    }
+
+    @Test
+    public void testWhenGetResultAndErrorThenLogged() throws Exception {
+        // Set up our test
+        class ErrorResponse extends TestResponse {
+            @Override
+            public InputStream getPayloadStream() {
+                String payload = "{\"" + OpenSearchWorkCoordinator.RESULT_OPENSSEARCH_FIELD_NAME + "\": \"slow your roll, dude\"}";
+                return new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+            }
+
+            @Override
+            public int getStatusCode() {
+                return 429;
+            }
+        }
+        Supplier<AbstractedHttpClient.AbstractHttpResponse> responseSupplier = () -> new ErrorResponse();
+        MockHttpClient client = new MockHttpClient(responseSupplier);
+
+        // Run the test & verify
+        DocumentModificationResult result;
+        try (
+                var workCoordinator = new OpenSearchWorkCoordinator(client, 2, "testWorker");
+                var closeableLogSetup = new CloseableLogSetup(workCoordinator.getClass().getName())) {
+            Assertions.assertThrows(IllegalArgumentException.class, () -> {
+                workCoordinator.getResult(responseSupplier.get());
+            });
+            System.out.println("Logged events: " + closeableLogSetup.getLogEvents());
```

Comment:
This should use logging or be removed.

Reply:
Done, think this got left in accidentally from testing the tests.